x-pack/plugins/fleet/server/services/agents/reassign.ts

```diff
@@ -113,16 +118,11 @@ export async function reassignAgents(
         throw new AgentReassignmentError(`${agent.id} is already assigned to ${newAgentPolicyId}`);
       }
 
-      const isAllowed = await reassignAgentIsAllowed(
-        soClient,
-        esClient,
-        agent.id,
-        newAgentPolicyId
-      );
-      if (isAllowed) {
-        return agent;
+      // by kuery already filtered out managed agents
+      if (!('kuery' in options)) {
+        await reassignAgentIsAllowed(soClient, esClient, agent.id, newAgentPolicyId);
```

Comment:
keeping the `reassignAgentIsAllowed` for other use cases when not using `kuery` (selecting up to 50 from UI)